rotkehlchen/globaldb/handler.py

```diff
@@ -298,35 +298,53 @@ def get_assets_mappings(identifiers: List[str]) -> Dict[str, dict]:
         return result
 
     @staticmethod
-    def search_assets(filter_query: 'AssetsFilterQuery') -> List[Dict[str, str]]:
+    def search_assets(filter_query: 'AssetsFilterQuery', db: 'DBHandler') -> List[Dict[str, str]]:
         """Returns a list of asset details that match the search query provided."""
         search_result = []
         query, bindings = GlobalDBHandler()._prepare_search_assets_query(filter_query)  # noqa:E501
-        with GlobalDBHandler().conn.read_ctx() as cursor:
-            cursor.execute(query, bindings)
-            for entry in cursor:
-                search_result.append({
-                    'identifier': entry[0],
-                    'name': entry[1],
-                    'symbol': entry[2],
-                })
+        with db.conn.read_ctx() as cursor:
+            with GlobalDBHandler().conn.read_ctx() as global_db_cursor:
+                global_db_cursor.execute(query, bindings)
+                found_eth = False
+                for entry in global_db_cursor:
+                    if db.get_settings(cursor).treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
+                        if found_eth is True:
+                            continue
+                        found_eth = True
+
+                    search_result.append({
+                        'identifier': entry[0],
+                        'name': entry[1],
+                        'symbol': entry[2],
+                    })
```

Comment:
I believe that if you find `ETH2` before `ETH` the `search_result` has `ETH2` and not `ETH`